src/metabase/query_processor/middleware/limit.clj

```diff
@@ -27,13 +30,26 @@
            (ensure-reduced result')
            result'))))))
 
+(defn determine-query-max-rows
```

Comment:
there's a pretty similar thing in `metabase.query-processor.middleware.add-rows-truncated` which should prolly be folded in